addons/website_form_builder/controllers/main.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+import base64
+
+import json
+
+from openerp import http, SUPERUSER_ID
+from openerp.http import request
+from openerp.tools.translate import _
+
+class form_builder(http.Controller):
+    @http.route('/website_form/thanks/page/<template>', type='http', auth="public", website=True)
+    def thanks_page(self, template):
+        template = ('website.' if template.split('.',1)[0] != 'website' else '')+template
+
+        try:
+            request.website.get_template(template)
+        except ValueError:
+            return request.redirect('/page/'+template)
+
+        model = request.session.get('form_builder_model')
+        record = request.env.sudo()[model].browse(request.session.get('form_builder_id'))
+        if not record.exists():
+            return False
+        return request.website.render(template, {"record":record})
+
+    @http.route('/website_form/<model>', type='http', auth="public", website=True)
+    def website_form(self, model, **kwargs):
+
+        form = request.env['website.form'].search([('model_name', '=', model)])
```

Comment:
if you need several form, you dont need to send its on the same time with same parameters. This part dont need to be managed on the controller.